### ISSUE TITLE
Microsoft.AppCenter.Crashes clang++ exited with code 1 when compiling for ios-simulator on pair arm64 mac in VS for Windows

### ISSUE THREAD

**Issue by jsiemensLatitudegeo** [microsoft/appcenter-sdk-dotnet#1755]
### **Description**

If you try to use the Microsoft.AppCenter.Crashes in a .net maui app in Visual Studio for Windows paired with an arm64 Mac, deploying to the ios simulator, then you get an error:

Severity	Code	Description	Project	File	Line	Suppression State
Error		clang++ exited with code 1:
ld: in /Users/[user]/Library/Caches/Xamarin/mtbs/builds/MauiApp1/baf8b3e6a4f75682439ae911f842ade66bd1741bc02fc255b853516d938283c3/obj/Debug/net7.0-ios/iossimulator-x64/linker-cache/AppCenter.a(MSACAppCenter.o), building for iOS Simulator, but linking in object file built for iOS, file '/Users/[user]/Library/Caches/Xamarin/mtbs/builds/MauiApp1/baf8b3e6a4f75682439ae911f842ade66bd1741bc02fc255b853516d938283c3/obj/Debug/net7.0-ios/iossimulator-x64/linker-cache/AppCenter.a' for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)	MauiApp1	C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\16.4.7098\targets\Xamarin.Shared.Sdk.targets	1274	

### **Repro Steps**

1. Create new maui app in VS for Windows
2. Pair to Mac (arm64 - eg. m1 or m2)
3. Build and deploy to ios simulator

### **Details**

1. What is your app platform (Xamarin.Android or Xamarin.iOS or UWP)?
    - e.g. MAUI on all 3 platforms (Android, iOS and UWP).
2. If using Xamarin.Forms or if using portable/shared code to call our SDK APIs, are you using shared project, PCL code or .NET standard code for the application? Which .NET standard version or which PCL profile?
    - e.g. Multitargeting
2. Which SDK version are you using?
    - e.g. 5.0.2
4. Which OS version did you experience the issue on?
    - e.g. iOS 16.4 Simulator
5. What device version did you see this error on?  Were you using an emulator or a physical device?
    - e.g. iPhone 14 Simulator running iOS 16.4
6. What third party libraries are you using? 
    - Just yours
7. Please enable verbose logging for your app using `AppCenter.LogLevel = LogLevel.Verbose` before your call to `AppCenter.Start(...)` and include the logs here:

**Comment by IBFAO** [comment]
I have the same error with the iOS simulator on Mac since I upgraded to .net 8

**Comment by FM1973** [comment]
I got the same problem.
I my case I´ve installed .net RC1 as well. But I got the problem when using AppCenter in .net 7 with MAUI 7.0.59.

**Comment by rogerbriggen-securiton** [comment]
Same here. After installing dotnet 8 RC1 on the mac I have the same problem when compiling for dotnet 8.
Note: I am building on the mac from the console so it is not about windows and pair but it is just the compilation on the mac.

**Comment by tipa** [comment]
Hope this will be fixed soon. Very easy to reproduce

```
dotnet new ios
dotnet add package Microsoft.AppCenter.Analytics
dotnet build
```

```
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/16.4.8825-net8-rc1/targets/Xamarin.Shared.Sdk.targets(1589,3): error : clang++ exited with code 1: [/Users/tipa/GitHub/Neuer Ordner/Neuer Ordner.csproj]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/16.4.8825-net8-rc1/targets/Xamarin.Shared.Sdk.targets(1589,3): error : ld: warning: ignoring duplicate libraries: '-lSystem.Globalization.Native', '-lSystem.IO.Compression.Native', '-lSystem.Native', '-lSystem.Net.Security.Native', '-lSystem.Security.Cryptography.Native.Apple', '-lcompression', '-lmono-component-debugger', '-lmono-component-diagnostics_tracing', '-lmono-component-hot_reload', '-lmono-component-marshal-ilgen', '-lmonosgen-2.0', '-lxamarin-dotnet-debug' [/Users/tipa/GitHub/Neuer Ordner/Neuer Ordner.csproj]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/16.4.8825-net8-rc1/targets/Xamarin.Shared.Sdk.targets(1589,3): error : ld: building for 'iOS-simulator', but linking in object file (/Users/tipa/GitHub/Neuer Ordner/obj/Debug/net8.0-ios/iossimulator-arm64/linker-cache/AppCenter.a[arm64][2](MSACAppCenter.o)) built for 'iOS' [/Users/tipa/GitHub/Neuer Ordner/Neuer Ordner.csproj]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/16.4.8825-net8-rc1/targets/Xamarin.Shared.Sdk.targets(1589,3): error : clang: error: linker command failed with exit code 1 (use -v to see invocation) [/Users/tipa/GitHub/Neuer Ordner/Neuer Ordner.csproj]
```

**Comment by vecalion** [comment]
Experiencing the same issue.

**Comment by DmitriyKirakosyan** [comment]
@tipa 

> Hope this will be fixed soon. Very easy to reproduce
> 
> dotnet new ios
> dotnet add package Microsoft.AppCenter.Analytics
> dotnet build


The build succeeded for me using your steps. Are you building it on intel or m1/m2 mac?

**Comment by tipa** [comment]
I am building on a M1 Mac. Make sure it creates and builds the project with .NET 8 - it builds as expected with .NET 7. I just tested again with .NET 8 RC 2 and the error still shows


```
tipa@Mac-mini asdad % dotnet build
MSBuild version 17.8.0+6cdef4241 for .NET
  Determining projects to restore...
  Restored /Users/tipa/GitHub/subtotal/SubTotal.macOS/asdad/asdad.csproj (in 133 ms).
/usr/local/share/dotnet/sdk/8.0.100-rc.2.23502.2/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.RuntimeIdentifierInference.targets(311,5): message NETSDK1057: You are using a preview version of .NET. See: https://aka.ms/dotnet-support-policy [/Users/tipa/GitHub/subtotal/SubTotal.macOS/asdad/asdad.csproj]
  Detected signing identity:
          
    Bundle Id: com.companyname.asdad
    App Id: com.companyname.asdad
  asdad -> /Users/tipa/GitHub/subtotal/SubTotal.macOS/asdad/bin/Debug/net8.0-ios/iossimulator-arm64/asdad.dll
  Optimizing assemblies for size may change the behavior of the app. Be sure to test after publishing. See: https://aka.ms/dotnet-illink
  Optimizing assemblies for size. This process might take a while.
  Tool xcrun execution finished (exit code = 1).
          
  ld: warning: ignoring duplicate libraries: '-lSystem.Globalization.Native', '-lSystem.IO.Compression.Native', '-lSystem.Native', '-lSystem.Net.Security.Native', '-lSystem.Security.Cryptography.Native.Apple', '-lcompression', '-lmono-component-debugger', '-lmono-component-diagnostics_tracing', '-lmono-component-hot_reload', '-lmono-component-marshal-ilgen', '-lmonosgen-2.0', '-lxamarin-dotnet-debug'
  ld: building for 'iOS-simulator', but linking in object file (/Users/tipa/GitHub/subtotal/SubTotal.macOS/asdad/obj/Debug/net8.0-ios/iossimulator-arm64/linker-cache/AppCenter.a[arm64][2](MSACAppCenter.o)) built for 'iOS'
  clang: error: linker command failed with exit code 1 (use -v to see invocation)
  
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/16.4.8968-net8-rc2/targets/Xamarin.Shared.Sdk.targets(1556,3): error : clang++ exited with code 1: [/Users/tipa/GitHub/subtotal/SubTotal.macOS/asdad/asdad.csproj]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/16.4.8968-net8-rc2/targets/Xamarin.Shared.Sdk.targets(1556,3): error : ld: warning: ignoring duplicate libraries: '-lSystem.Globalization.Native', '-lSystem.IO.Compression.Native', '-lSystem.Native', '-lSystem.Net.Security.Native', '-lSystem.Security.Cryptography.Native.Apple', '-lcompression', '-lmono-component-debugger', '-lmono-component-diagnostics_tracing', '-lmono-component-hot_reload', '-lmono-component-marshal-ilgen', '-lmonosgen-2.0', '-lxamarin-dotnet-debug' [/Users/tipa/GitHub/subtotal/SubTotal.macOS/asdad/asdad.csproj]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/16.4.8968-net8-rc2/targets/Xamarin.Shared.Sdk.targets(1556,3): error : ld: building for 'iOS-simulator', but linking in object file (/Users/tipa/GitHub/subtotal/SubTotal.macOS/asdad/obj/Debug/net8.0-ios/iossimulator-arm64/linker-cache/AppCenter.a[arm64][2](MSACAppCenter.o)) built for 'iOS' [/Users/tipa/GitHub/subtotal/SubTotal.macOS/asdad/asdad.csproj]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/16.4.8968-net8-rc2/targets/Xamarin.Shared.Sdk.targets(1556,3): error : clang: error: linker command failed with exit code 1 (use -v to see invocation) [/Users/tipa/GitHub/subtotal/SubTotal.macOS/asdad/asdad.csproj]

Build FAILED.

```

**Comment by DmitriyKirakosyan** [comment]
That's intriguing. For Mac users, the issue appears only on .NET 8. However, when building from Windows in conjunction with Mac, it's reproducible on .NET 7.

If the problem stemmed from a missing architecture in the AppCenter SDK binary, it would be present across all versions of .NET. I suspect this might be a .NET-related issue. Let's keep this open and gather additional details about this problem.

If anyone has a solution or suggestion, contributions are always appreciated!

**Comment by rolfbjarne** [comment]
> If the problem stemmed from a missing architecture in the AppCenter SDK binary, it would be present across all versions of .NET. I suspect this might be a .NET-related issue. Let's keep this open and gather additional details about this problem.

It's an AppCenter SDK issue, the problem is that the AppCenter SDK doesn't support arm64 in the simulator.

The difference is that in later versions .NET/IDE we've changed the default architecture (RuntimeIdentifier) we build for the simulator when building on an arm64 mac: in initial .NET 6 + .NET 7 releases the default was iossimulator-x64, now (in .NET 8 + some versions of .NET 7 depending on the VS version) it's iossimulator-arm64 when building on an arm64 mac.

That's why the workaround works in the IDE:

```xml
<PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net7.0-ios|AnyCPU'">
    <ForceSimulatorX64ArchitectureInIDE>true</ForceSimulatorX64ArchitectureInIDE>
</PropertyGroup>
```

It forces using `iossimulator-x64` when building from the IDE.

Note that device builds are not affected in any way.

**Comment by markbeaton** [comment]
This has been biting us for a while too as we much prefer to develop using the ARM64 Simulator on our Apple Silicon Macs, but it's currently not possible for dotnet projects - we are either forced to use the x64 sim via Rosetta, or remove references to the AppCenter packages.

I suspect a lot of folks don't realise that ARM64 simulator and ARM64 device binaries are two different, non-compatible things & support for both requires bundling XCFrameworks inside the nupkg payloads, containing both the `ios-arm64` and `ios-arm64-simulator` (or `ios-arm64_x86_64-simulator`) sub-frameworks.

[See here for more information](https://github.com/xamarin/xamarin-macios/issues/17841) (including lots of good stuff from @rolfbjarne)

**Comment by Hooterr** [comment]
It's not a deal breaker, but it'd certainly be nice to be able to run arm64 simulator. When run in x64 mode, the simulator is (imo) a bit slower, and there's a scrolling bug. You can't fling a scrollview or collection (it's an iOS bug but MAUI, same happens in Swift app).

**Comment by sledderGIT** [comment]
Experiencing the same issue. Have to exclude Appcenter from my Solution.

**Comment by plppp2001** [comment]
> Same issue when upgrading MAUI workloads from `7.0.86` to `7.0.92` as well as from .NET SDK `7.0.302` to `7.0.401`, on both Windows and paired Mac. Pretty sure my Xcode on the ARM64 Mac was already on 14.3 and this was working just fine before the .NET upgrades. I also updated the AppCenter nuget from `5.0.2` to `5.0.3`, but same results.
> 
> Ultimately the suggestion by the MAUI team [on another thread](https://github.com/dotnet/maui/issues/16778#issuecomment-1683343255) allowed me to build and deploy to an iOS simulator, but not sure if there are any side-effects yet. Have not yet tried on a physical device, or release deployment for that matter.
> 
> Quick fix:
> 
> ```
> <PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net7.0-ios|AnyCPU'">
>     <ForceSimulatorX64ArchitectureInIDE>true</ForceSimulatorX64ArchitectureInIDE>
> </PropertyGroup>
> ```
> 
> For the MS-AppCenter team, the MAUI team suggested the following for the [reason](https://github.com/dotnet/maui/issues/16778#issuecomment-1680958629) for the issue:
> 
> > @rolfbjarne: Looks like AppCenter doesn't ship support for the iossimulator-arm64 architecture, so this is an AppCenter issue.

I tried this doesn't work, but I'm using .NET8 I did this:

	<PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net8.0-ios|AnyCPU'">
		<ForceSimulatorX64ArchitectureInIDE>true</ForceSimulatorX64ArchitectureInIDE>
	</PropertyGroup>

**Comment by plppp2001** [comment]
> Experiencing the same issue. Have to exclude Appcenter from my Solution.

.NET 8 MAUI?

**Comment by daniel-smith-sonepar** [comment]
agreed i have same issue with .net 8 maui and workaround is not working for me either

**Comment by plppp2001** [comment]
> agreed i have same issue with .net 8 maui and workaround is not working for me either

Can you try this out for me please, add this setting, reload your project, close VS and reopen and build again?

	<PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|**net8.0-ios**|AnyCPU'">
		<ApplicationIdGuid>D1A6A67F-D8FB-4D43-BAA6-908DEBC59120</ApplicationIdGuid>
		<ForceSimulatorX64ArchitectureInIDE>true</ForceSimulatorX64ArchitectureInIDE>
		<RuntimeIdentifier>iossimulator-x64</RuntimeIdentifier>
	</PropertyGroup>

**Comment by daniel-smith-sonepar** [comment]
actually i just had same thought - i simply unloaded project, reloaded and did a rebuild and it launched!

**Comment by plppp2001** [comment]
> actually i just had same thought - i simply unloaded project, reloaded and did a rebuild and it launched!
> 
> So the workaround did work for me once i did reload and rebuild of project.

Did you add this one -->  <RuntimeIdentifier>iossimulator-x64</RuntimeIdentifier>

**Comment by daniel-smith-sonepar** [comment]
> > actually i just had same thought - i simply unloaded project, reloaded and did a rebuild and it launched!
> > So the workaround did work for me once i did reload and rebuild of project.
> 
> Did you add this one: `<RuntimeIdentifier>iossimulator-x64</RuntimeIdentifier>`

No I did not add that line and I confirmed I do not have the RuntimeIdentifier set in the project file anywhere

**Comment by kocevilija** [comment]
> > agreed i have same issue with .net 8 maui and workaround is not working for me either
> 
> Can you try this out for me please, add this setting, reload your project, close VS and reopen and build again?
> 
> ```
> <PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|**net8.0-ios**|AnyCPU'">
> 	<ForceSimulatorX64ArchitectureInIDE>true</ForceSimulatorX64ArchitectureInIDE>
> 	<RuntimeIdentifier>iossimulator-x64</RuntimeIdentifier>
> </PropertyGroup>
> ```

This solution has worked for me to fix the issue after updating to .NET8 and MAUI.

PS. I am using both [Microsoft.AppCenter.Crashes](https://www.nuget.org/packages/Microsoft.AppCenter.Crashes) and [BarcodeScanner.Mobile.Maui](https://www.nuget.org/packages/BarcodeScanner.Mobile.Maui) nuget packages that do not have the arm64 slice and they work as well.

**Comment by sledderGIT** [comment]
For me this 

> > > agreed i have same issue with .net 8 maui and workaround is not working for me either
> > 
> > 
> > Can you try this out for me please, add this setting, reload your project, close VS and reopen and build again?
> > ```
> > <PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|**net8.0-ios**|AnyCPU'">
> > 	<ForceSimulatorX64ArchitectureInIDE>true</ForceSimulatorX64ArchitectureInIDE>
> > 	<RuntimeIdentifier>iossimulator-x64</RuntimeIdentifier>
> > </PropertyGroup>
> > ```
> 
> This solution has worked for me to fix the issue after updating to .NET8 and MAUI 8.
> 
> PS. I am using both [Microsoft.AppCenter.Crashes](https://www.nuget.org/packages/Microsoft.AppCenter.Crashes) and [BarcodeScanner.Mobile.Maui](https://www.nuget.org/packages/BarcodeScanner.Mobile.Maui) nuget packages that do not have the arm64 slice and they work as well.

I'm working on Mac with M1 so I want to launch my project with ARM64 simulator. So for me this workaround is not suetable.

**Comment by DmitriyKirakosyan** [comment]
I'm going to upload the POC nuget packages, which are intended to resolve this issue. If you have some time, could you please test it and let me know if it works for you?

[Microsoft.AppCenter.5.0.5-SNAPSHOT.nupkg.zip](https://github.com/microsoft/appcenter-sdk-dotnet/files/13739580/Microsoft.AppCenter.5.0.5-SNAPSHOT.nupkg.zip)
[Microsoft.AppCenter.Analytics.5.0.5-SNAPSHOT.nupkg.zip](https://github.com/microsoft/appcenter-sdk-dotnet/files/13739582/Microsoft.AppCenter.Analytics.5.0.5-SNAPSHOT.nupkg.zip)
[Microsoft.AppCenter.Crashes.5.0.5-SNAPSHOT.nupkg.zip](https://github.com/microsoft/appcenter-sdk-dotnet/files/13739583/Microsoft.AppCenter.Crashes.5.0.5-SNAPSHOT.nupkg.zip)

**Comment by sledderGIT** [comment]
> I'm going to upload the POC nuget packages, which are intended to resolve this issue. If you have some time, could you please test it and let me know if it works for you?
> 
> [Microsoft.AppCenter.5.0.5-SNAPSHOT.nupkg.zip](https://github.com/microsoft/appcenter-sdk-dotnet/files/13739580/Microsoft.AppCenter.5.0.5-SNAPSHOT.nupkg.zip) [Microsoft.AppCenter.Analytics.5.0.5-SNAPSHOT.nupkg.zip](https://github.com/microsoft/appcenter-sdk-dotnet/files/13739582/Microsoft.AppCenter.Analytics.5.0.5-SNAPSHOT.nupkg.zip) [Microsoft.AppCenter.Crashes.5.0.5-SNAPSHOT.nupkg.zip](https://github.com/microsoft/appcenter-sdk-dotnet/files/13739583/Microsoft.AppCenter.Crashes.5.0.5-SNAPSHOT.nupkg.zip)

Sure! I'll test it

**Comment by snckirkmarken** [comment]
Hi @DmitriyKirakosyan, this looks like it's worked for us so far, thank you!

The same issue exists with `Microsoft.AppCenter.Distribute`. Do you plan to add the same change for that too?

**Comment by haavamoa** [comment]
```   
 <PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|**net8.0-ios**|AnyCPU'">
        <ForceSimulatorX64ArchitectureInIDE>true</ForceSimulatorX64ArchitectureInIDE>
        <RuntimeIdentifier>iossimulator-x64</RuntimeIdentifier>
</PropertyGroup>
```

Did not work for us, MAUI .NET 8. :(

**Comment by markbeaton** [comment]
@DmitriyKirakosyan Confirming that the `5.0.5-SNAPSHOT` packages are working for me on the ARM64 Simulator - thanks.

**Comment by haavamoa** [comment]
> > agreed i have same issue with .net 8 maui and workaround is not working for me either
> 
> Can you try this out for me please, add this setting, reload your project, close VS and reopen and build again?
> 
> ```
> <PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|**net8.0-ios**|AnyCPU'">
> 	<ForceSimulatorX64ArchitectureInIDE>true</ForceSimulatorX64ArchitectureInIDE>
> 	<RuntimeIdentifier>iossimulator-x64</RuntimeIdentifier>
> </PropertyGroup>
> ```

The workaround to set RuntimeIdentifier to x64 is not feasible as you now have to manually switch between `iossimulator-x64` and `ios-arm64` when developing for phyiscal device / simulator :(

**Comment by DmitriyKirakosyan** [comment]
> The same issue exists with Microsoft.AppCenter.Distribute. Do you plan to add the same change for that too?

@snckirkmarken , Distribute is a bit trickier. Working on it...

**Comment by jbe2277** [comment]
I believe this issue is because of the reference: [SQLitePCLRaw.bundle_green](https://www.nuget.org/packages/SQLitePCLRaw.bundle_green/) (>= 2.1.5) in the [AppCenter 5.0.3](https://www.nuget.org/packages/Microsoft.AppCenter/5.0.3#dependencies-body-tab) package.

The SQLitePCL.raw repo fixed this in 2.1.6:
- https://github.com/ericsink/SQLitePCL.raw/releases/tag/v2.1.6

Workaround:
- https://github.com/jbe2277/waf/commit/3aa8ea358ced2d1b26f57fe66f4b6b01a9a472ae

**Comment by SprengerS** [comment]
> I'm going to upload the POC nuget packages, which are intended to resolve this issue. If you have some time, could you please test it and let me know if it works for you?
> 
> [Microsoft.AppCenter.5.0.5-SNAPSHOT.nupkg.zip](https://github.com/microsoft/appcenter-sdk-dotnet/files/13739580/Microsoft.AppCenter.5.0.5-SNAPSHOT.nupkg.zip) [Microsoft.AppCenter.Analytics.5.0.5-SNAPSHOT.nupkg.zip](https://github.com/microsoft/appcenter-sdk-dotnet/files/13739582/Microsoft.AppCenter.Analytics.5.0.5-SNAPSHOT.nupkg.zip) [Microsoft.AppCenter.Crashes.5.0.5-SNAPSHOT.nupkg.zip](https://github.com/microsoft/appcenter-sdk-dotnet/files/13739583/Microsoft.AppCenter.Crashes.5.0.5-SNAPSHOT.nupkg.zip)

The packages are working on the new iOS Simulator Devices, but not on real Devices. The following error in the console occurs: 
```
ASI found [dyld] (sensitive) 'Library not loaded: @rpath/Frameworks/AppCenterCrashes.framework/AppCenterCrashes
  Referenced from: <6BD34BA9-9317-35E1-89A2-BB1CAC37D4D1> /private/var/containers/Bundle/Application/691CA60F-FEA0-4A98-9585-A1E62E2A3C62/Hoermann.BleApp.iOS.app/Hoermann.BleApp.iOS
  Reason: tried: '/private/var/containers/Bundle/Application/691CA60F-FEA0-4A98-9585-A1E62E2A3C62/Hoermann.BleApp.iOS.app/Frameworks/Frameworks/AppCenterCrashes.framework/AppCenterCrashes' (no such file), '/private/var/containers/Bundle/Application/691CA60F-FEA0-4A98-9585-A1E62E2A3C62/Hoermann.BleApp.iOS.app/Frameworks/Frameworks/AppCenterCrashes.framework/AppCenterCrashes' (no such file)'
```

**Comment by durandt** [comment]
> > > > agreed i have same issue with .net 8 maui and workaround is not working for me either
> > > 
> > > 
> > > Can you try this out for me please, add this setting, reload your project, close VS and reopen and build again?
> > > ```
> > > <PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|**net8.0-ios**|AnyCPU'">
> > > 	<ForceSimulatorX64ArchitectureInIDE>true</ForceSimulatorX64ArchitectureInIDE>
> > > 	<RuntimeIdentifier>iossimulator-x64</RuntimeIdentifier>
> > > </PropertyGroup>
> > > ```
> > 
> > 
> > This solution has worked for me to fix the issue after updating to .NET8 and MAUI 8.
> > PS. I am using both [Microsoft.AppCenter.Crashes](https://www.nuget.org/packages/Microsoft.AppCenter.Crashes) and [BarcodeScanner.Mobile.Maui](https://www.nuget.org/packages/BarcodeScanner.Mobile.Maui) nuget packages that do not have the arm64 slice and they work as well.
> 
> I'm working on Mac with M1 so I want to launch my project with ARM64 simulator. So for me this workaround is not suetable.
> 
> We have just cut the AppCenter from Debug configs
> 
> ```
>   <ItemGroup Condition="!$(Configuration.Contains('Debug'))">
>     <PackageReference Include="Microsoft.AppCenter" Version="5.0.3" />
>     <PackageReference Include="Microsoft.AppCenter.Analytics" Version="5.0.3" />
>     <PackageReference Include="Microsoft.AppCenter.Crashes" Version="5.0.3" />
>   </ItemGroup>
> ```

Same here, work-arounds would not make the error disappear (M1 Pro mac, tried using Rider, VSCode and command line build)

**Comment by InquisitorJax** [comment]
any eta on when the 5.0.5 fix will be released?

**Comment by snckirkmarken** [comment]
> > The same issue exists with Microsoft.AppCenter.Distribute. Do you plan to add the same change for that too?
> 
> @snckirkmarken , Distribute is a bit trickier. Working on it...

Happy New Year @DmitriyKirakosyan. Please may I ask: how are the `Microsoft.AppCenter.Distribute` changes coming along?

**Comment by magtimmermans** [comment]
I am waiting for the release. Do you know when the 5.0.5 fix will be released?

**Comment by snckirkmarken** [comment]
@plppp2001 that workaround doesn't work for us

**Comment by plppp2001** [comment]
Report that then bill gates

www.PaulAstramowicz.com
________________________________

**Comment by Mercally** [comment]
> I'm going to upload the POC nuget packages, which are intended to resolve this issue. If you have some time, could you please test it and let me know if it works for you?
> 
> [Microsoft.AppCenter.5.0.5-SNAPSHOT.nupkg.zip](https://github.com/microsoft/appcenter-sdk-dotnet/files/13739580/Microsoft.AppCenter.5.0.5-SNAPSHOT.nupkg.zip) [Microsoft.AppCenter.Analytics.5.0.5-SNAPSHOT.nupkg.zip](https://github.com/microsoft/appcenter-sdk-dotnet/files/13739582/Microsoft.AppCenter.Analytics.5.0.5-SNAPSHOT.nupkg.zip) [Microsoft.AppCenter.Crashes.5.0.5-SNAPSHOT.nupkg.zip](https://github.com/microsoft/appcenter-sdk-dotnet/files/13739583/Microsoft.AppCenter.Crashes.5.0.5-SNAPSHOT.nupkg.zip)

Thank you @DmitriyKirakosyan 

I tested with prerelease version provided, it seems it doesn't get fatal errors, but still doesn't work as expected on iOS.

Test environment: MacBook Pro M1. iPhone 15 iOS 17.2 Simulator. (debug mode)
Dev environment: Visual Code/Visual Studio for Mac. .Net 8.0 MAUI 8

Issue: No analytics/events collected on App Center portal.
Just registering the AppCenter.Start(...) at App.xaml.cs class. And added Analytics.TrackEvent(...) at HomePage OnAppearing.

The debug console:
![Screenshot 2024-01-16 at 10 18 02 AM](https://github.com/microsoft/appcenter-sdk-dotnet/assets/29028686/d7b0a874-7072-413f-8995-8cc711ef5f9d)

Note: It seems there are no issues on Android.

**Comment by magtimmermans** [comment]
I have tested on a Windows 10 Machine, Visual Studio 2022 Community. .Net 8.0 MAUI 8. 

Here it is also not crashing but also no Analytics in AppCenter. Basicly the same as @Mercally

**Comment by cleardemon** [comment]
Just to add my experiences here, 5.0.3 (current release version on NuGet) works just fine in deployments, on device. I am not using Windows, but macOS as my development machine (Rider and VSCode). 

I have tried using the 5.0.5 snapshots, but, while they allow for the iOS Simulator builds to compile, attempting to use them on an actual device (and test deployment) doesn't work too well, and also fails at the signing process in release builds.

My workaround for now is to continue to use 5.0.3 release versions in deployments, but completely remove App Center from builds on local developer machines. I achieved this by setting a global build constant (by adding a `DefineConstants` property to [Directory.Build.props](https://learn.microsoft.com/en-us/visualstudio/msbuild/customize-by-directory)), some changes to my `.csproj` files to exclude referencing any App Center packages, and `#ifdef` blocks in the code to exclude any calls into `AppCenter` in the compile.

**Comment by nathenxbrewer** [comment]
This is stopping me from debugging on physical Android devices. I'm using Rider. I'm not entirely sure why Rider is failing to deploy to an Android device for an ios-simulator issue..I've tried all the above workarounds and nothing changes. Using an M3 MacBook Pro.

**Comment by durandt** [comment]
@nathenxbrewer yeee it's a pretty big bummer. Doesn't help though that Rider seems to have its own set of (unfixed) issues with that (I'm second-guessing, not properly applying the csproj properties suggested in the fixes here).
See https://youtrack.jetbrains.com/issue/RIDER-102647/.NET-8-MAUI-Build-fails-on-rider-but-works-fine-on-VS-for-Mac

I was able to make it work (see my message from 3 weeks ago) when building from command-line and from VSCode but I still get the error on Rider. Sad because the debugging experience in VSCode is horrendous right now (compared to what was possible with VS/VS for mac/Rider)

**Comment by nathenxbrewer** [comment]
> @nathenxbrewer yeee it's a pretty big bummer. Doesn't help though that Rider seems to have its own set of (unfixed) issues with that (I'm second-guessing, not properly applying the csproj properties suggested in the fixes here). See https://youtrack.jetbrains.com/issue/RIDER-102647/.NET-8-MAUI-Build-fails-on-rider-but-works-fine-on-VS-for-Mac
> 
> I was able to make it work (see my message from 3 weeks ago) when building from command-line and from VSCode but I still get the error on Rider. Sad because the debugging experience in VSCode is horrendous right now (compared to what was possible with VS/VS for mac/Rider)

If I disable Linking, I can deploy to physical Android, but I need to be able to use Hot Reload to work on the UI in my app. The MAUI kit for VSCode is decent, but it also doesn't support Hot Reload. I'll take your suggestion and use that ItemGroup as well as add some compiler directive statements to exclude the using statements. I just got the M3 2 days ago, this was all working great on my Intel Mac I had before that.

**Comment by nathenxbrewer** [comment]
> @nathenxbrewer yeee it's a pretty big bummer. Doesn't help though that Rider seems to have its own set of (unfixed) issues with that (I'm second-guessing, not properly applying the csproj properties suggested in the fixes here). See https://youtrack.jetbrains.com/issue/RIDER-102647/.NET-8-MAUI-Build-fails-on-rider-but-works-fine-on-VS-for-Mac
> 
> I was able to make it work with the fixes suggested above when building from command-line and from VSCode but I still get the error on Rider. Sad because the debugging experience in VSCode is horrendous right now (compared to what was possible with VS/VS for mac/Rider).
> 
> Don't have access to code right now but I think this fix did it:
> 
> ```
>   <ItemGroup Condition="!$(Configuration.Contains('Debug'))">
>     <PackageReference Include="Microsoft.AppCenter" Version="5.0.3" />
>     <PackageReference Include="Microsoft.AppCenter.Analytics" Version="5.0.3" />
>     <PackageReference Include="Microsoft.AppCenter.Crashes" Version="5.0.3" />
>   </ItemGroup>
> ```

Now it builds on Rider, but my app insta-crashes on both iOS and Android now. I think I'm just gonna run Visual Studio through Parallels and see if that does any better.

**Comment by alikoli** [comment]
I have 2 probles with these packages:
[Microsoft.AppCenter.5.0.5-SNAPSHOT.nupkg.zip](https://github.com/microsoft/appcenter-sdk-dotnet/files/13739580/Microsoft.AppCenter.5.0.5-SNAPSHOT.nupkg.zip)
[Microsoft.AppCenter.Analytics.5.0.5-SNAPSHOT.nupkg.zip](https://github.com/microsoft/appcenter-sdk-dotnet/files/13739582/Microsoft.AppCenter.Analytics.5.0.5-SNAPSHOT.nupkg.zip)
[Microsoft.AppCenter.Crashes.5.0.5-SNAPSHOT.nupkg.zip](https://github.com/microsoft/appcenter-sdk-dotnet/files/13739583/Microsoft.AppCenter.Crashes.5.0.5-SNAPSHOT.nupkg.zip)

1. It is not possible to send app to app store. When trying to publish to AppStore I get an error:
Invalid Bundle. The bundle at 'myapp.app/Frameworks/AppCenterCrashes.framework' contains disallowed file 'Frameworks'

2. I was testing this package with development on Windows with local device (ipad os). After moving to this package from previous 5.0.3 version I was able to use AppCenter.Start method, which previously caused an error. Calling method Crashes.TrackError with these packages does produce app crash.

**Comment by dmn-star** [comment]
@DmitriyKirakosyan 
My app can be built with 5.0.5-SNAPSHOT packages, runs in the iOS simulator but crashes immediately on the  iPhone.

**Comment by ryanalford-suzy** [comment]
Any update on when this will actually get fixed? For me, the workaround when running on a Mac, worked in .Net 7. Now that we have updated to .Net 8, the workaround no longer works. I am back to getting the same build error.

This has been an issue for more than 4 months.

**Comment by DmitriyKirakosyan** [comment]
Hello everyone, and thank you for your patience.

I'd like to outline the necessary steps to address this issue. Our approach involves using xcframeworks in our binding projects. Specifically, this requires building the [AppCenter Apple SDK](https://github.com/microsoft/appcenter-sdk-apple) as a dynamic framework, a challenging task given the complexity of the project's structure. At present, we distribute static xcframeworks, and while .NET 8 might support them (this requires verification), they are not compatible with .NET 7.

Regrettably, I cannot provide a specific ETA for this resolution, as it has been deferred in favor of other priorities. As a temporary solution, I recommend debugging on a physical device or excluding AppCenter during build for emulator debugging.

If you're open to experimenting with a solution for this issue, your efforts would be greatly appreciated. We value and welcome all assistance and contributions.

**Comment by markbeaton** [comment]
>  As a temporary solution, I recommend debugging on a physical device or excluding AppCenter during build for emulator debugging.

Here's what we're doing at the moment to achieve this:
```xml
<ItemGroup>
	<PackageReference Include="Microsoft.Extensions.Logging.Debug" Version="8.0.0"/>
	<!-- other packages etc...-->
</ItemGroup>
<ItemGroup Condition=" '$(Configuration)' != 'Debug' ">
	<PackageReference Include="Microsoft.AppCenter.Analytics" Version="5.0.3"/>
	<PackageReference Include="Microsoft.AppCenter.Crashes" Version="5.0.3"/>
</ItemGroup>
```

Something along these lines at app startup:
```c#
private void InitialiseAppCenter()
{
	#if !DEBUG
	var appCenterServices = new List<Type>();
	if (BuildConfig.AppCenterCrashesEnabled)
	{
		appCenterServices.Add(typeof(Microsoft.AppCenter.Crashes.Crashes));
	}
	if (BuildConfig.AppCenterAnalyticsEnabled)
	{
		appCenterServices.Add(typeof(Microsoft.AppCenter.Analytics.Analytics));
	}
	if (appCenterServices.Count > 0)
	{
		Microsoft.AppCenter.AppCenter.Start(BuildConfig.AppCenterSecret, appCenterServices.ToArray());
	}
	#endif
}
```

**Comment by AlexeyStarkov** [comment]
> > If the problem stemmed from a missing architecture in the AppCenter SDK binary, it would be present across all versions of .NET. I suspect this might be a .NET-related issue. Let's keep this open and gather additional details about this problem.
> 
> It's an AppCenter SDK issue, the problem is that the AppCenter SDK doesn't support arm64 in the simulator.
> 
> The difference is that in later versions .NET/IDE we've changed the default architecture (RuntimeIdentifier) we build for the simulator when building on an arm64 mac: in initial .NET 6 + .NET 7 releases the default was iossimulator-x64, now (in .NET 8 + some versions of .NET 7 depending on the VS version) it's iossimulator-arm64 when building on an arm64 mac.
> 
> That's why the workaround works in the IDE:
> 
> ```
> <PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net7.0-ios|AnyCPU'">
>     <ForceSimulatorX64ArchitectureInIDE>true</ForceSimulatorX64ArchitectureInIDE>
> </PropertyGroup>
> ```
> 
> It forces using `iossimulator-x64` when building from the IDE.
> 
> Note that device builds are not affected in any way.

this workaround works for me when debugging on iOS using .net8 and Rider IDE. However, the issue comes back when I target an Android device. I mean, c'mon, why does iOS-related crap involve Android builds at all?!!! I've been trying to perform a simple update from .net7 to .net8 for the last two days. And I haven't even seen yet if the application actually looks/behaves as it's expected because it doesn't build. 

Does anyone know how to ignore/suppress this issue while building and deploying to Android? 

PS: This is just insane - I'm asking for a workaround for a workaround.

**Comment by durandt** [comment]
@AlexeyStarkov This is also blocking me from updating my MAUI app to .NET 8 and I have sticked to .NET 7. But as I submitted an update to my app yesterday, I was reminded by Apple that I cannot do that for too long:

> We noticed one or more issues with a recent delivery for the following app:
> 
> _&lt;Redacted app information&gt;_
> 
> Although delivery was successful, you may want to correct the following issues in your next delivery. Once you've corrected the issues, upload a new binary to App Store Connect.
> 
> ITMS-90725: SDK version issue - This app was built with the iOS 16.4 SDK. Starting April 29, 2024, all iOS and iPadOS apps must be built with the iOS 17 SDK or later, included in Xcode 15 or later, in order to be uploaded to App Store Connect or submitted for distribution.

I am looking for alternatives to App Center for crash reporting and logging a few events (anyone reading this may give suggestions).

@AlexeyStarkov be aware that Rider has its own set of issues regarding MAUI and .NET 8...
See https://youtrack.jetbrains.com/issue/RIDER-76794/Cannot-deploy-MAUI-project-to-physical-iOS-device-because-of-RuntimeIdentifier (not sure if the right issue, jetbrains issue tracker does not load right now, will update later)
So I would advise you to try using VSCode with the C# DevKit and MAUI extensions (although it's not really easy to set up and comes with its own load of issues...). I was able to build with VSCode where Rider would still throw errors.

**Comment by AlexeyStarkov** [comment]
@durandt valid guess about VSCode. Definitely worth to try this Lego-like notepad on steroids finally. Hope preparing it for work will take less than another two days.

When it comes to staying with .net7, upgrading to .net8 is my try to run away from other bugs/issues I have lost hope of overcoming.

**Comment by durandt** [comment]
I agree we really should be upgrading to .NET 8 asap. Was just mentioning that this issue is a pretty big blocker.

**Comment by dmn-star** [comment]
@AlexeyStarkov 
> However, the issue is still there when I target the Android. I mean, c'mon, why does iOS-related crap impact Android builds at all?!!! 

Because there is not really an iOS or Android build here, just an all-in-one build for all targets.
You can temporary delete net8.0-ios from the line in your project, if you only want to work on android part.
 <TargetFrameworks>net8.0;net8.0-android;net8.0-ios</TargetFrameworks>

**Comment by asierpn** [comment]
We had the same issue and with these project settings it works fine in both Debug and Release in our .NET8 iOS app (not MAUI).

```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>net8.0-ios</TargetFramework>
    <OutputType>Exe</OutputType>
    <ImplicitUsings>true</ImplicitUsings>
    <SupportedOSPlatformVersion>15.0</SupportedOSPlatformVersion>
    <EnableAssemblyILStripping>False</EnableAssemblyILStripping>
    <ProvisioningType>manual</ProvisioningType>
  </PropertyGroup>
  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
    <RuntimeIdentifier>iossimulator-x64</RuntimeIdentifier>
    <ForceSimulatorX64ArchitectureInIDE>true</ForceSimulatorX64ArchitectureInIDE>
    <MtouchLink>SdkOnly</MtouchLink>
    <CodesignKey>XXXX</CodesignKey>
    <CodesignProvision>XXXX</CodesignProvision>
  </PropertyGroup>
  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|AnyCPU'">
    <RuntimeIdentifier>ios-arm64</RuntimeIdentifier>
    <MtouchLink>SdkOnly</MtouchLink>
    <CodesignKey>XXXX</CodesignKey>
    <CodesignProvision>XXXX</CodesignProvision>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="Microsoft.Maui.Essentials" Version="8.0.6" />
    <PackageReference Include="Microsoft.AppCenter.Analytics" Version="5.0.3" />
    <PackageReference Include="Microsoft.AppCenter.Crashes" Version="5.0.3" />
    <PackageReference Include="SQLitePCLRaw.bundle_green" Version="2.1.7" />
  </ItemGroup>
</Project>
```

**Comment by AlexeyStarkov** [comment]
> We had the same issue and with these project settings it works fine in both Debug and Release in our .NET8 iOS app (not MAUI).
> 
> ```
> <Project Sdk="Microsoft.NET.Sdk">
>   <PropertyGroup>
>     <TargetFramework>net8.0-ios</TargetFramework>
>     <OutputType>Exe</OutputType>
>     <ImplicitUsings>true</ImplicitUsings>
>     <SupportedOSPlatformVersion>15.0</SupportedOSPlatformVersion>
>     <EnableAssemblyILStripping>False</EnableAssemblyILStripping>
>     <ProvisioningType>manual</ProvisioningType>
>   </PropertyGroup>
>   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
>     <RuntimeIdentifier>iossimulator-x64</RuntimeIdentifier>
>     <ForceSimulatorX64ArchitectureInIDE>true</ForceSimulatorX64ArchitectureInIDE>
>     <MtouchLink>SdkOnly</MtouchLink>
>     <CodesignKey>XXXX</CodesignKey>
>     <CodesignProvision>XXXX</CodesignProvision>
>   </PropertyGroup>
>   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|AnyCPU'">
>     <RuntimeIdentifier>ios-arm64</RuntimeIdentifier>
>     <MtouchLink>SdkOnly</MtouchLink>
>     <CodesignKey>XXXX</CodesignKey>
>     <CodesignProvision>XXXX</CodesignProvision>
>   </PropertyGroup>
>   <ItemGroup>
>     <PackageReference Include="Microsoft.Maui.Essentials" Version="8.0.6" />
>     <PackageReference Include="Microsoft.AppCenter.Analytics" Version="5.0.3" />
>     <PackageReference Include="Microsoft.AppCenter.Crashes" Version="5.0.3" />
>     <PackageReference Include="SQLitePCLRaw.bundle_green" Version="2.1.7" />
>   </ItemGroup>
> </Project>
> ```

Your config looks promising because the following code snippet is the one that must resolve the problem of whatever target platform is chosen.

```
<PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
    <RuntimeIdentifier>iossimulator-x64</RuntimeIdentifier>
    <ForceSimulatorX64ArchitectureInIDE>true</ForceSimulatorX64ArchitectureInIDE>
</PropertyGroup>
```

however, I tried it yesterday and when the target platform is Android it leads to another error:
"Microsoft.Android.Sdk.Tooling.targets(58, 5): [XA0035] Unable to determine the Android ABI from the value 'iossimulator-x64'. Edit the project file in a text editor and set the 'RuntimeIdentifiers' MSBuild property to contain only valid identifiers for the Android platform."

It makes sense of course, from the build system's point of view. But from my point of view, it means I need to look for a workaround for a workaround of a workaround.

**Comment by ryanalford-suzy** [comment]
> 
> Your config looks promising because the following code snippet is the one that must resolve the problem no matter what target platform is chosen.
> 
> ```
> <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
>     <RuntimeIdentifier>iossimulator-x64</RuntimeIdentifier>
>     <ForceSimulatorX64ArchitectureInIDE>true</ForceSimulatorX64ArchitectureInIDE>
> </PropertyGroup>
> ```
> 
> however, I tried it yesterday and when the target platform is Android it leads to another error: "Microsoft.Android.Sdk.Tooling.targets(58, 5): [XA0035] Unable to determine the Android ABI from the value 'iossimulator-x64'. Edit the project file in a text editor and set the 'RuntimeIdentifiers' MSBuild property to contain only valid identifiers for the Android platform."
> 
> It makes sense of course, from the build system's point of view. But from my point of view, it means I need to look for a workaround for a workaround of a workaround.

If you want to do iOS specific project configurations, you can use this...

```
<PropertyGroup Condition="$(TargetFramework.StartsWith('net8.0-ios')) != true AND $(TargetFramework.StartsWith('net8.0-maccatalyst')) != true">

</PropertyGroup>
```

Anything inside would be for Debug or Release on iOS or MacCatalyst. If you aren't supporting mac, then you can take that part off.

However, since we are talking about the simulator workaround mainly, this would probably work for you...

```
<PropertyGroup Condition="'$(Configuration)|$(TargetFramework)'=='Debug|net8.0-ios'">
	<RuntimeIdentifier>iossimulator-x64</RuntimeIdentifier>
    <ForceSimulatorX64ArchitectureInIDE>true</ForceSimulatorX64ArchitectureInIDE>
</PropertyGroup>
```

**Comment by ryanalford-suzy** [comment]
> > > Your config looks promising because the following code snippet is the one that must resolve the problem no matter what target platform is chosen.
> > > ```
> > > <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
> > >     <RuntimeIdentifier>iossimulator-x64</RuntimeIdentifier>
> > >     <ForceSimulatorX64ArchitectureInIDE>true</ForceSimulatorX64ArchitectureInIDE>
> > > </PropertyGroup>
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > however, I tried it yesterday and when the target platform is Android it leads to another error: "Microsoft.Android.Sdk.Tooling.targets(58, 5): [XA0035] Unable to determine the Android ABI from the value 'iossimulator-x64'. Edit the project file in a text editor and set the 'RuntimeIdentifiers' MSBuild property to contain only valid identifiers for the Android platform."
> > > It makes sense of course, from the build system's point of view. But from my point of view, it means I need to look for a workaround for a workaround of a workaround.
> > 
> > 
> > If you want to do iOS specific project configurations, you can use this...
> > ```
> > <PropertyGroup Condition="$(TargetFramework.StartsWith('net8.0-ios')) != true AND $(TargetFramework.StartsWith('net8.0-maccatalyst')) != true">
> > 
> > </PropertyGroup>
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Anything inside would be for Debug or Release on iOS or MacCatalyst. If you aren't supporting mac, then you can take that part off.
> > However, since we are talking about the simulator workaround mainly, this would probably work for you...
> > ```
> > <PropertyGroup Condition="'$(Configuration)|$(TargetFramework)'=='Debug|net8.0-ios'">
> > 	<RuntimeIdentifier>iossimulator-x64</RuntimeIdentifier>
> >     <ForceSimulatorX64ArchitectureInIDE>true</ForceSimulatorX64ArchitectureInIDE>
> > </PropertyGroup>
> > ```
> 
> no, as I told above, it doesn't work (at least to me) because both Android and iOS are built no matter the target platform is chosen and iOS simulator error appears even when you try to build an app for android.

You project code from a couple posts above is putting the simulator runtime identifier for any DEBUG build...so it will be included for both iOS and Android. So yeah, I could see you getting your issue because you are telling the build system to include the simulator runtime identifier on android also.

**Comment by ryanalford-suzy** [comment]
> Yes, I tried to do that because the builder throws the same clang++ iossimulator error no matter what platform I try to build and deploy to.

Spent a little more time on it this morning and this now works for me on .Net 8 running on a Mac with Rider IDE. Built and deployed to iOS simulator, iPhone, and Android phone. Omit the other options if they aren't necessary for you.

```
<PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net8.0-ios|AnyCPU'">
  <CreatePackage>false</CreatePackage>
  <CodesignProvision>Automatic</CodesignProvision>
  <CodesignEntitlements>Platforms\iOS\Entitlements.development.plist</CodesignEntitlements>
  <!-- Fixes for AppCenter nuget package to build/run with simulators -->
  <RuntimeIdentifier>iossimulator-x64</RuntimeIdentifier>
  <ForceSimulatorX64ArchitectureInIDE>true</ForceSimulatorX64ArchitectureInIDE>
</PropertyGroup>
```

**Comment by dmn-star** [comment]
@AlexeyStarkov
> Sorry. It was a rhetorical question caused by Xamarin -> MAUI migration frustration.

Np,  I can understand your frustration quite well.  Anyway, please upvote to save others the trouble in the future. :-)
https://youtrack.jetbrains.com/issue/RIDER-91070/Running-iOS-Run-Configuration-also-Builds-other-TargetFrameworks

**Comment by AlexeyStarkov** [comment]
> > As a temporary solution, I recommend debugging on a physical device or excluding AppCenter during build for emulator debugging.
> 
> Here's what we're doing at the moment to achieve this (app `.csproj` file):
> 
> ```
> <ItemGroup>
> 	<PackageReference Include="Microsoft.Extensions.Logging.Debug" Version="8.0.0"/>
> 	
> </ItemGroup>
> <ItemGroup Condition=" '$(Configuration)' != 'Debug' ">
> 	<PackageReference Include="Microsoft.AppCenter.Analytics" Version="5.0.3"/>
> 	<PackageReference Include="Microsoft.AppCenter.Crashes" Version="5.0.3"/>
> </ItemGroup>
> ```
> 
> Something along these lines at app startup:
> 
> ```cs
> private void InitialiseAppCenter()
> {
> 	#if !DEBUG
> 	var appCenterServices = new List<Type>();
> 	if (BuildConfig.AppCenterCrashesEnabled)
> 	{
> 		appCenterServices.Add(typeof(Microsoft.AppCenter.Crashes.Crashes));
> 	}
> 	if (BuildConfig.AppCenterAnalyticsEnabled)
> 	{
> 		appCenterServices.Add(typeof(Microsoft.AppCenter.Analytics.Analytics));
> 	}
> 	if (appCenterServices.Count > 0)
> 	{
> 		Microsoft.AppCenter.AppCenter.Start(BuildConfig.AppCenterSecret, appCenterServices.ToArray());
> 	}
> 	#endif
> }
> ```

I've ended up with the workaround shown above eventually. It seems to be the only reliable solution :(

**Comment by durandt** [comment]
Oh well...

![image](https://github.com/microsoft/appcenter-sdk-dotnet/assets/1219650/a8e93ef2-81d6-4ff7-b91a-e0ba83239df4)

Visual Studio App Center is scheduled for retirement on March 31, 2025. [Learn more about support timelines and recommended alternatives.](https://aka.ms/appcenter/retire)

**Comment by MitchBomcanhao** [comment]
It doesn't look like any of the recommend alternatives are related to mobile app crash/error reporting and telemetry?

**Comment by durandt** [comment]
> It doesn't look like any of the recommend alternatives are related to mobile app crash/error reporting and telemetry?

I would also like to hear about alternatives for crash/error reporting and telemetry. 
@jfversluis did a video om Sentry: https://www.youtube.com/watch?v=9-50zH8fqYA
But it seems to be uploading waaaaaay too much data for my liking (sure, it's neat to debug, but you have to consider your users privacy too)

**Comment by bruno-garcia** [comment]
The goal is to automatically attach data that can be useful for debugging problems. Device model etc. What data do you find that affects user privacy? 

Worth noting you have control of what’s sent. You get the whole object in a callback where you can delete data before it leaves the device. And you can also add filtering rules to run at ingestion time before the data hits disk.

**Comment by durandt** [comment]
@bruno-garcia thank you for chipping in, really appreciate it!

One more question while you are, how about one of my comments on Mastodon, that the .NET Sentry SDK seems to (from what I see in the video) be sending crash reports directly rather than saving on disk to be sent at next app startup?
Docs for unhandled exceptions for iOS and Android both suggest the code handling exceptions should be quick about it, and in my mind it has always been excluded to do some network stuff there.
So my question is: Does the .NET Sentry SDK attempt to do network operations in the unhandled exception code? If it fail, will it save it to the disk and retry at startup? Can this behavior be configured?
Thanks!

**Comment by durandt** [comment]
Regarding the amount of data: 
The couple of minutes after 11:20 in this video: https://youtu.be/9-50zH8fqYA?feature=shared&t=680
Thats: 
. Stacktrace
. List of rendering events
. about 30-40 data points*
. loaded assemblies

Sure that could be interesting but that seems like a huge footprint to add to every events (I like to add _many_ events and usually just want to track a couple of values associated with the event)

* data points
![image](https://github.com/microsoft/appcenter-sdk-dotnet/assets/1219650/9c8409a3-34da-4626-9ebb-a560a787c45b)

**Comment by bruno-garcia** [comment]
You ended up not pointing out what is concerning regarding user privacy there. I see that as technical detail potentially useful for debugging.  

Regarding amount, data is compressed before leaving the device and it’s not a lot specially since it’s only in the event of an error. 

Ultimately as I said you can control what’s sent, so you can keep only the stack trace if you feel that’s right for you.

**Comment by bruno-garcia** [comment]
> the .NET Sentry SDK seems to (from what I see in the video) be sending crash reports directly rather than saving on disk to be sent at next app startup?

It stores it on disk first and deletes it after sending it. 
In the event of a native crash it can only work this way. It can’t make network calls out of a signal handler.

> Docs for unhandled exceptions for iOS and Android both suggest the code handling exceptions should be quick about it, and in my mind it has always been excluded to do some network stuff there.

On Android it tries to send it, but only after writing to disk. In my tests if a network connection was already open, for Java exceptions it managed to send immediately. But NDK crashes always only after restart. 

> So my question is: Does the .NET Sentry SDK attempt to do network operations in the unhandled exception code? If it fails, will it save it to the disk and retry at startup? Can this behavior be configured?

C#, Java, Koltin, Swift, C++, Objective-C (any native) crashes will store crash data on disk first. In managed code, it will **try** to send it immediately (because it can work, not a real OS signal handler , it’s a managed code global error handler). On the native crashes only on app reload data is sent out

**Comment by durandt** [comment]
Thank you for your explanations!

> You ended up not pointing out what is concerning regarding user privacy there. I see that as technical detail potentially useful for debugging.

There aren't really specific data-points that are directly affecting user privacy, but the combination of some of them _may_ increase the risk of making it possible to identify some users.
Battery level -> Could be ranges or "IsBatteryLevel10PercentOrLower" = true/false
Storage/memory size -> Could be "HasAtLeast100MBFreeStorage" = true/false (100MB or another threshold)

Regarding footprint, there is also a (minor) concern that the more data you fetch using platform APIs, the bigger risk that these any of these APIs get deprecated and start creating issues (Keep it stupide simple)

I am not trying to be negative toward Sentry, I may very well use it in the end. But I have burnt myself in the past using 3:rd party crashlytics tools that made my apps crash and I am just being very cautious.

**Comment by markuspalme** [comment]
Would you mind taking your conversation regarding sentry somewhere private?

**Comment by durandt** [comment]
Apologies. I still think other developers may be interested by the discussion so I created #1772 for further discussions about how to move away from AppCenter since it is now announced as retiring in a year.

**Comment by DmitriyKirakosyan** [comment]
As we do not have plans to support net8 and fix this bug in the next year, I'm closing the issue.

**Comment by markuspalme** [comment]
@thomasgalliker It works, just not in the simulator. You can conditionally disable it in the simulator.

**Comment by thomasgalliker** [comment]
@markuspalme this is if your car mechanic tells you "your car runs fine, just on three wheels". We do most of our development work on iOS simulator because it's by far the fastest way to get a compiled result on a phone.

**Comment by thisisthekap** [comment]
Just as note to help fellow developers stumbling uppon this issue: Modify your csproj file to force your ios simulator into x64 as given here: https://developercommunity.visualstudio.com/t/Unable-to-build--MAUI-app-for-iOS-with-x/10501186#TPIN-N10501850

**Comment by jvondermarck** [comment]
### What solved the issue for me was to setup like this my `csproj` file

```csproj
<Project Sdk="Microsoft.NET.Sdk">
	<PropertyGroup>
		<TargetFrameworks>net8.0-android;net8.0-ios</TargetFrameworks>
		<OutputType>Exe</OutputType>
		<SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'ios'">11.0</SupportedOSPlatformVersion>
		<SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'android'">21.0</SupportedOSPlatformVersion>
                  ...
	</PropertyGroup>

	<!-- ⚠️ Workaround on mac to be able to run the project on Android and iOS -->
	<PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net8.0-ios|AnyCPU'">
		<RuntimeIdentifier>iossimulator-x64</RuntimeIdentifier>
		<ForceSimulatorX64ArchitectureInIDE>true</ForceSimulatorX64ArchitectureInIDE>
	</PropertyGroup>

        <ItemGroup Condition="'$(TargetFramework)' == 'net8.0-android'">
		<ProjectReference Include="..\<Android lib>\<Android lib>.csproj" />
	</ItemGroup>

	<ItemGroup Condition="'$(TargetFramework)' == 'net8.0-ios'">
	        <ProjectReference Include="..\<iOS lib>\<iOS lib>.csproj" />
	</ItemGroup>

         ...

</Project>
```

-----

> [!NOTE] 
> The weird part is that I got the `0>Xamarin.Shared.Sdk.targets(1641,3): Error : clang++ exited with code 1:` error when I was running my project on Android, while on iOS it worked amazingly, but anyway thanks for the workaround !  (This error only happened on my M1 Mac, while on Windows it was working great without this workaround. 🤓
